core/src/Cabana_NeighborList.hpp

```diff
@@ -108,12 +110,67 @@ maxNeighbor( const ListType& list, const std::size_t num_particles )
 {
     std::size_t max_n = 0;
     for ( std::size_t p = 0; p < num_particles; p++ )
+    {
         if ( NeighborList<ListType>::numNeighbor( list, p ) > max_n )
             max_n = NeighborList<ListType>::numNeighbor( list, p );
+    }
     return max_n;
 }
 } // namespace Impl
 
+/*!
+  \brief Construct a histogram of neighbors per particle.
+  \param list Neighbor list with valid NeighborList interface.
+  \param num_bin Number of bins for the histogram.
+  \return Neighbor list histogram.
+*/
+template <class ExecutionSpace, class ListType>
+Kokkos::View<int* [2], typename ListType::memory_space>
+neighborHistogram( ExecutionSpace exec_space, const std::size_t num_particles,
+                   const ListType& list, const int num_bin )
+{
+    // Allocate View of neighbors per particle
+    auto num_neigh = Kokkos::View<int*, typename ListType::memory_space>(
+        "particle_count", num_particles );
+
+    // Extract from neighbor list interface.
+    auto extract_functor = KOKKOS_LAMBDA( const int p )
+    {
+        num_neigh( p ) = NeighborList<ListType>::numNeighbor( list, p );
+    };
+    Kokkos::RangePolicy<ExecutionSpace> particle_policy( exec_space, 0,
+                                                         num_particles );
+    Kokkos::parallel_for( particle_policy, extract_functor );
+    Kokkos::fence();
+
+    auto bin_data = Cabana::binByKey( num_neigh, num_bin );
+
+    auto histogram = Kokkos::View<int* [2], typename ListType::memory_space>(
+        "particle_histogram", num_bin, 2 );
+
+    int max_neigh;
+    Kokkos::Max<int> max_reduce( max_neigh );
+    Kokkos::parallel_reduce(
```

Comment:
I setup this reduce similar to the way `maxNeighbor` is called in `neighbor_unit_test.hpp` but it looks like we may be double looping over particles this way. We loop over particles in the `parallel_reduce` and then again in `Impl::maxNeighbor`. This version appears to work but we should probably look at changing the `maxNeighbor` interface so that we don't have this double loop